### PR TITLE
Update to 'v0.34.27' of github.com:cometbft/cometbft

### PR DESCRIPTION
**Pull request by cmwaters**
Closes https://github.com/celestiaorg/celestia-core/issues/961

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # global owners are only requested if there isn't a more specific
 # codeowner specified below. For this reason, the global codeowners
 # are often repeated in package-level definitions.
-*       @evan-forbes @Wondertan
+    *     @evan-forbes @Wondertan
```
Comment (evan-forbes) [inline .github/CODEOWNERS:10]: reminder to myself that we should update these in a different PR

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -3,5 +3,29 @@
...
+
+-->
+
+---
+
+#### PR checklist
+
+- [ ] Tests written/updated
+- [ ] Changelog entry added in `.changelog` (we use
+  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
```
Comment (evan-forbes) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: we can start using this if we want, but we might want to remove this if not

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -3,5 +3,29 @@
...
+https://github.com/orgs/cometbft/projects/1
+
+-->
+
+---
+
+#### PR checklist
+
+- [ ] Tests written/updated
+- [ ] Changelog entry added in `.changelog` (we use
```
Comment (evan-forbes) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: same with other comment, usually we just use github's generate feature when publishing a new release in all of the other repos, but if we want to start using this I'm not against it

File: .github/issue_template.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+labels: needs-triage
+---
+
+<!--
+
+If you want to ask a general question, please create a new discussion instead of
+an issue: https://github.com/cometbft/cometbft/discussions
```
Comment (evan-forbes) [inline .github/issue_template.md:8]: I still thinking about if we should keep these refs or not. For cometbft related issues, referencing upstream definitely makes sense, but I'm not sure about all issues.

File: .github/workflows/check-generated.yml [hunk 5]
```diff
@@ -7,45 +7,43 @@ name: Check generated code
 on:
   pull_request:
     branches:
-      - v0.34.x-celestia
+      - main
+      - v0.34.x
```
Comment (evan-forbes) [inline .github/workflows/check-generated.yml:11]: do we need to revert this change?

File: .github/workflows/cometbft-docker.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+name: Docker
+# Rebuilds the CometBFT docker image on every push to main and creation of tags
+# and pushes the image to https://hub.docker.com/r/cometbft/cometbft
+on:
+  push:
+    branches:
+      - main
+      - v0.34.x
```
Comment (evan-forbes) [inline .github/workflows/cometbft-docker.yml:8]: same as above

File: .github/workflows/coverage.yml [hunk 7]
```diff
@@ -3,8 +3,7 @@ on:
   pull_request:
   push:
     branches:
-      - v[0-9]+.[0-9]+.x-celestia
-      - release/**
+      - v0.34.x
```
Comment (evan-forbes) [inline .github/workflows/coverage.yml:6]: same as above

File: abci/client/socket_client_test.go [hunk 8]
```diff
@@ -12,7 +12,7 @@ import (
 	abcicli "github.com/tendermint/tendermint/abci/client"
 	"github.com/tendermint/tendermint/abci/server"
 	"github.com/tendermint/tendermint/abci/types"
-	tmrand "github.com/tendermint/tendermint/libs/rand"
+	cmtrand "github.com/tendermint/tendermint/libs/rand"
```
Comment (evan-forbes) [inline abci/client/socket_client_test.go:15]: omg lol

File: docs/qa/v034/img/baseline/total_txs_rate_regular.png
Comment (evan-forbes) [inline docs/qa/v034/img/baseline/total_txs_rate_regular.png:1]: would be cool to eventually re-run these, but when using celestia-app, and maybe in a more celestia irl evironment

**Review by evan-forbes — COMMENTED** [review]
upstream changing all the relative imports to tm -> cmt seems a bit excessive imo lololol, but thanks for chugging through all of that!

glad the .vuepress stuff if gone from the docs. we should stop getting a bunch of javascript realted vulnribility warnings

File: .github/CODEOWNERS (hunk 1 above)
Comment (cmwaters) [reply .github/CODEOWNERS:10]: I think they're updated in the spec change pr

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (cmwaters) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: Yeah we don't have to but it might be annoying if we have two means of reporting changes within the same repo

File: abci/client/socket_client_test.go (hunk 8 above)
Comment (cmwaters) [reply abci/client/socket_client_test.go:15]: Yeah they went proper

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (cmwaters) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: I guess we could just continue to point to the cometbft changelog as well

**Comment by evan-forbes** [comment]
is the failing mock check related to #917? if so, I think we can perform the same lobotamy and merge

File: .github/CODEOWNERS (hunk 1 above)
Comment (rootulp) [reply .github/CODEOWNERS:10]: FYI the spec PR targets celestia-app not celestia-core

**Comment by cmwaters** [comment]
> is the failing mock check related to https://github.com/celestiaorg/celestia-core/pull/917? if so, I think we can perform the same lobotamy and merge
> 
> the other vulnerability check might be because its using v0.19.6 instead of v0.19.7 like on upstream, but idk

I think it's because the file is incorrect and `mockery` isn't being installed. But if I correct it, I think we will experience the same failure as you allude to.

The go vuln CI is new. You're probably right about the go version because I think it's patched after go 1.19.7. I'm not sure if we can force the CI to use a later version or if it just updates it later

**Review by evan-forbes — APPROVED** [review]
